### PR TITLE
Fix submit filters

### LINKED ISSUE #6865: Advanced filter value lost when changing default value type
#### Sonata packages

<details><summary>show</summary>
<p>

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.89.0 3.89.0 The missing Symfony Admin Generator
sonata-project/block-bundle              4.5.0  4.5.0  Symfony SonataBlockBundle
sonata-project/cache                     2.0.1  2.1.1  Cache library
sonata-project/datagrid-bundle           3.3.0  3.3.0  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.11.0 1.11.0 Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.29.0 3.29.0 Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/exporter                  2.5.1  2.5.1  Lightweight Exporter library
sonata-project/form-extensions           1.9.0  1.9.0  Symfony form extensions
sonata-project/twig-extensions           1.5.1  1.5.1  Sonata twig extensions
sonata-project/user-bundle               4.9.0  4.11.0 Symfony SonataUserBundle

```

</p>
</details>

#### Symfony packages

<details><summary>show</summary>
<p>

```
$ composer show --latest 'symfony/*'
symfony/apache-pack                v1.0.1  v1.0.1  A pack for Apache support in Symfony
symfony/asset                      v4.4.19 v5.2.3  Manages URL generation and versioning of web assets such as CSS stylesheets, JavaScript files and image files
symfony/browser-kit                v4.4.19 v5.2.3  Simulates the behavior of a web browser, allowing you to make requests, click on links and submit forms programm...
symfony/cache                      v4.4.19 v5.2.3  Provides an extended PSR-6, PSR-16 (and tags) implementation
symfony/cache-contracts            v2.2.0  v2.2.0  Generic abstractions related to caching
symfony/config                     v4.4.19 v5.2.3  Helps you find, load, combine, autofill and validate configuration values of any kind
symfony/console                    v4.4.19 v5.2.3  Eases the creation of beautiful and testable command line interfaces
symfony/css-selector               v4.4.19 v5.2.3  Converts CSS selectors to XPath expressions
symfony/debug                      v4.4.19 v4.4.19 Provides tools to ease debugging PHP code
symfony/debug-bundle               v4.4.19 v5.2.3  Provides a tight integration of the Symfony Debug component into the Symfony full-stack framework
symfony/debug-pack                 v1.0.9  v1.0.9  A debug pack for Symfony projects
symfony/dependency-injection       v4.4.19 v5.2.3  Allows you to standardize and centralize the way objects are constructed in your application
symfony/deprecation-contracts      v2.2.0  v2.2.0  A generic function and convention to trigger deprecation notices
symfony/doctrine-bridge            v4.4.19 v5.2.3  Provides integration for Doctrine with various Symfony components
symfony/dom-crawler                v4.4.19 v5.2.3  Eases DOM navigation for HTML and XML documents
symfony/dotenv                     v4.4.19 v5.2.3  Registers environment variables from a .env file
symfony/error-handler              v4.4.19 v5.2.3  Provides tools to manage errors and ease debugging PHP code
symfony/event-dispatcher           v4.4.19 v5.2.3  Provides tools that allow your application components to communicate with each other by dispatching events and l...
symfony/event-dispatcher-contracts v1.1.9  v2.2.0  Generic abstractions related to dispatching event
symfony/expression-language        v4.4.19 v5.2.3  Provides an engine that can compile and evaluate expressions
symfony/filesystem                 v4.4.19 v5.2.3  Provides basic utilities for the filesystem
symfony/finder                     v4.4.19 v5.2.3  Finds files and directories via an intuitive fluent interface
symfony/flex                       v1.12.1 v1.12.1 Composer plugin for Symfony
symfony/form                       v4.4.19 v5.2.3  Allows to easily create, process and reuse HTML forms
symfony/framework-bundle           v4.4.19 v5.2.3  Provides a tight integration between Symfony components and the Symfony full-stack framework
symfony/http-client-contracts      v2.3.1  v2.3.1  Generic abstractions related to HTTP clients
symfony/http-foundation            v4.4.19 v5.2.3  Defines an object-oriented layer for the HTTP specification
symfony/http-kernel                v4.4.19 v5.2.3  Provides a structured process for converting a Request into a Response
symfony/inflector                  v4.4.19 v5.2.3  Converts words between their singular and plural forms (English only)
symfony/intl                       v4.4.19 v5.2.3  Provides a PHP replacement layer for the C intl extension that includes additional data from the ICU library
symfony/maker-bundle               v1.29.1 v1.29.1 Symfony Maker helps you create empty commands, controllers, form classes, tests and more so you can forget about...
symfony/mime                       v4.4.19 v5.2.3  Allows manipulating MIME messages
symfony/monolog-bridge             v4.4.19 v5.2.3  Provides integration for Monolog with various Symfony components
symfony/monolog-bundle             v3.6.0  v3.6.0  Symfony MonologBundle
symfony/options-resolver           v4.4.19 v5.2.3  Provides an improved replacement for the array_replace PHP function
symfony/orm-pack                   v1.2.0  v2.1.0  A pack for the Doctrine ORM
symfony/phpunit-bridge             v5.2.3  v5.2.3  Provides utilities for PHPUnit, especially user deprecation notices management
symfony/polyfill-intl-grapheme     v1.22.1 v1.22.1 Symfony polyfill for intl's grapheme_* functions
symfony/polyfill-intl-icu          v1.22.1 v1.22.1 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn          v1.22.1 v1.22.1 Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-intl-normalizer   v1.22.1 v1.22.1 Symfony polyfill for intl's Normalizer class and related functions
symfony/polyfill-mbstring          v1.22.1 v1.22.1 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72             v1.22.1 v1.22.1 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73             v1.22.1 v1.22.1 Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/polyfill-php80             v1.22.1 v1.22.1 Symfony polyfill backporting some PHP 8.0+ features to lower PHP versions
symfony/profiler-pack              v1.0.5  v1.0.5  A pack for the Symfony web profiler
symfony/property-access            v4.4.19 v5.2.3  Provides functions to read and write from/to an object or array using a simple string notation
symfony/property-info              v4.4.19 v5.2.3  Extracts information about PHP class' properties using metadata of popular sources
symfony/routing                    v4.4.19 v5.2.3  Maps an HTTP request to a set of configuration variables
symfony/security-acl               v3.1.1  v3.1.1  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle            v4.4.19 v5.2.3  Provides a tight integration of the Security component into the Symfony full-stack framework
symfony/security-core              v4.4.19 v5.2.3  Symfony Security Component - Core Library
symfony/security-csrf              v4.4.19 v5.2.3  Symfony Security Component - CSRF Library
symfony/security-guard             v4.4.19 v5.2.3  Symfony Security Component - Guard
symfony/security-http              v4.4.19 v5.2.3  Symfony Security Component - HTTP Integration
symfony/serializer                 v4.4.19 v5.2.3  Handles serializing and deserializing data structures, including object graphs, into array structures or other f...
symfony/service-contracts          v2.2.0  v2.2.0  Generic abstractions related to writing services
symfony/stopwatch                  v4.4.19 v5.2.3  Provides a way to profile code
symfony/string                     v5.2.3  v5.2.3  Provides an object-oriented API to strings and deals with bytes, UTF-8 code points and grapheme clusters in a un...
symfony/swiftmailer-bundle         v3.5.2  v3.5.2  Symfony SwiftmailerBundle
symfony/templating                 v4.4.19 v5.2.3  Provides all the tools needed to build any kind of template system
symfony/translation                v4.4.19 v5.2.3  Provides tools to internationalize your application
symfony/translation-contracts      v2.3.0  v2.3.0  Generic abstractions related to translation
symfony/twig-bridge                v4.4.19 v5.2.3  Provides integration for Twig with various Symfony components
symfony/twig-bundle                v4.4.19 v5.2.3  Provides a tight integration of Twig into the Symfony full-stack framework
symfony/validator                  v4.4.19 v5.2.3  Provides tools to validate values
symfony/var-dumper                 v4.4.19 v5.2.3  Provides mechanisms for walking through any arbitrary PHP variable
symfony/var-exporter               v4.4.19 v5.2.3  Allows exporting any serializable PHP data structure to plain PHP code
symfony/web-profiler-bundle        v4.4.19 v5.2.3  Provides a development tool that gives detailed information about the execution of any request
symfony/webpack-encore-bundle      v1.11.0 v1.11.0 Integration with your Symfony app & Webpack Encore!
symfony/yaml                       v4.4.19 v5.2.3  Loads and dumps YAML files
```

</p>
</details>

#### PHP version

```
$ php -v
PHP 7.4.10 (cli) (built: Sep  1 2020 16:52:21) ( NTS Visual C++ 2017 x64 )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.10, Copyright (c), by Zend Technologies
```

## Subject

I have a admin list view with a default filter with activated advanced filter.

Here is my code to define the filters and default filter:

<details><summary>show</summary>
<p>

```
protected function configureDatagridFilters(DatagridMapper $datagridMapper)
    {
        $datagridMapper
            ->add('chatState',
                'doctrine_orm_choice', [
                    'global_search' => true,
                    'field_type'    => ChoiceType::class,
                    'field_options' => [
                        'choices' => ChatStates::getChoices()
                    ]
                ]
            );
    }

protected function configureDefaultFilterValues(array &$filterValues)
    {
        $filterValues['chatState'] = [
            'type'  => EqualOperatorType::TYPE_NOT_EQUAL,
            'value' => 'archived',
        ];
    }
```

</p>
</details>

When I change the filter type to 'equal' and press the 'Filter' the list should filter all items with chatState = 'archived'

Tested with sonata-admin verion 3.89. Have to step back to <3.86 to get old filter behavior.

## Expected results

- List is filtered with `chatState = 'archived'`
- url params should look like this: `list?filter[chatState][type]=1&filter[chatState][value]=archived`

## Actual results

- List is empty
- url params are: `list?filter[chatState][type]=1`
-> value is missing

### PR DESCRIPTION
**Pull request by kirya-dev**
Closes #6865.

### Changelog

```markdown
### Fixed
- Fix lost filter value when only type submitted
- Allow submit empty value when default filter non empty
- Invisible disabling filters (using remove name)
```

### CODE REVIEW

File: src/Resources/public/Admin.js [hunk 1]
```diff
@@ -782,6 +776,13 @@ var Admin = {
             }
         });
     },
+    convert_query_string_to_object: function (str) {
```
Comment (VincentLanglet) [inline src/Resources/public/Admin.js:779]: Can you add a newline before and after this function ?

**Comment by VincentLanglet** [comment]
I made multiple try with this code and I confirm that it seems to fix every issues. Nice work.

**Review by wbloszyk — APPROVED** [review]
For me looks good, anyway I am not good in js.

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
I totally forgot this wasn't working for filter with array values...

I'll wait for https://github.com/sonata-project/SonataAdminBundle/pull/6877 to make a release

**Pull request merged**